mkdocs/plugins.py

```diff
@@ -510,3 +510,50 @@ def run_event(self, name: str, item=None, **kwargs):
             if result is not None:
                 item = result
         return item
+
+
+class _PrefixLogger(logging.LoggerAdapter):
+    """A logger adapter to prefix log messages."""
+
+    def __init__(self, prefix: str, logger: logging.Logger):
+        """
+        Initialize the logger adapter.
+
+        Arguments:
+            prefix: The string to insert in front of every message.
+            logger: The logger instance.
+        """
+        super().__init__(logger, {})
+        self.prefix = prefix
+
+    def process(self, msg: str, kwargs: MutableMapping[str, Any]) -> tuple[str, Any]:
+        """
+        Process the message.
+
+        Arguments:
+            msg: The message:
+            kwargs: Remaining arguments.
+
+        Returns:
+            The processed message.
+        """
+        return f"{self.prefix}: {msg}", kwargs
+
+
+def get_plugin_logger(name: str) -> _PrefixLogger:
+    """Return a logger for plugins.
+
+    Arguments:
+        name: The name to use with `logging.getLogger`.
+
+    Returns:
+        A logger configured to work well in MkDocs,
+            prefixing each message with the plugin package name.
+
+    Examples:
+        >>> from mkdocs.plugins import get_plugin_logger
+        >>> logger = get_plugin_logger(__name__)
+        >>> logger.info("My plugin message")
```

Comment:
Hm MkDocs never uses `>>>` in examples, I'd prefer to keep it that way

Reply:
Python code block then?

Reply:
Yes